cpp/api/include/trtorch/trtorch.h

```diff
@@ -306,16 +307,45 @@ struct TRTORCH_API ExtraInfo {
      */
     bool strict_types = false;
 
-    /**
-     * (Only used when targeting DLA (device))
-     * Lets engine run layers on GPU if they are not supported on DLA
+
+    /*
+     * Setting data structure for Target device
      */
-    bool allow_gpu_fallback = true;
+    struct Device {
+	/**
+	 * @brief Setting data structure for device
+	 * This struct will hold Target device related parameters such as device_type, gpu_id, dla_core
+	 */
+	DeviceType device_type;
+
+	/*
+	 * Target gpu id
+	 */
+	uint64_t gpu_id;
+
+        /*
+         * When using DLA core on NVIDIA AGX platforms gpu_id should be set as Xavier device
+         */
+	int dla_core;
```

Comment:
Spacing alignment and do we want to call it dla_id so its the same as gpu_id? or is dla_core standard nomenclature?

Reply:
I tried aligning it with TensorRT's DLA nomenclature.